base/linalg/generic.jl

```diff
@@ -842,10 +855,11 @@ function (\)(A::AbstractMatrix, B::AbstractVecOrMat)
     return qrfact(A,Val(true)) \ B
 end
 
-(\)(a::AbstractVector, b::AbstractArray) = reshape(a, length(a), 1) \ b
+(\)(a::AbstractVector, b::AbstractArray) = pinv(a) * b
 (/)(A::AbstractVecOrMat, B::AbstractVecOrMat) = (B' \ A')'
 # \(A::StridedMatrix,x::Number) = inv(A)*x Should be added at some point when the old elementwise version has been deprecated long enough
 # /(x::Number,A::StridedMatrix) = x*inv(A)
+/(x::Number, v::AbstractVector) = x*pinv(v)
```

Comment:
This changes the result between two versions without any deprecation and have just caught me completely by surprise.

One more evidence why giving completely different meaning for operations that used to have elwize meanings makes no sense.

Reply:
Which versions? It is an error on 0.6.

Reply:
Ah, there used to be `/(x::Number, r::Range)`:
```julia
julia> VERSION
v"0.6.0"

julia> 1 / (1:4)
4-element Array{Float64,1}:
 1.0     
 0.5     
 0.333333
 0.25
```
```julia
julia> VERSION
v"0.7.0-DEV.2158"

julia> 1 / (1:4)
1×4 RowVector{Float64,Array{Float64,1}}:
 0.0333333  0.0666667  0.1  0.133333
```
The old method was removed in #22932 without adding an appropriate deprecation. So I guess we could add a deprecated `/(x::Number, r::Range)` to still do the element-wise operation, which would be more specific than the method defined here. @yuyichao is that the case that was bothering you?

Reply:
If I may add, when dividing a scalar by a vector, taking the pseudo-inverse of the "vector" and then multiplying it by a constant seems very far from an intuitive behavior to me. I think if one is doing this operation and one is aware it is taking the pseudoinverse, then explicitly calling `pinv` seems more appropriate rather than relying on this (in)convenience.